src/controllers/therapists.ts

```diff
@@ -0,0 +1,22 @@
+import { NextFunction, Request, Response } from 'express';
+import getAllTherapist from '../services';
+
+const getAllTherapists = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const { q, page } = req.query;
+    const pageNumber = parseInt(page as string, 2) || 1;
+    const limit = 8;
+
+    const therapists = await getAllTherapist(q as string || ' ', pageNumber, limit);
+
+    if (!therapists.length) {
```

Comment:
what if we have zero length therapist, What do u expect to happen,
Can u please trace the whole flow